future-apps/payroll/contracts/Payroll.sol

```diff
@@ -53,30 +53,37 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
     string private constant ERROR_EXPIRY_TIME_TOO_SHORT = "PAYROLL_EXPIRY_TIME_TOO_SHORT";
     string private constant ERROR_EXCHANGE_RATE_ZERO = "PAYROLL_EXCHANGE_RATE_ZERO";
     string private constant ERROR_PAST_TERMINATION_DATE = "PAYROLL_PAST_TERMINATION_DATE";
+    string private constant ERROR_LAST_PAYROLL_DATE_TOO_BIG = "PAYROLL_LAST_DATE_TOO_BIG";
 
     struct Employee {
         address accountAddress; // unique, but can be changed over time
-        mapping(address => uint8) allocation;
+        mapping(address => uint256) allocation;
         uint256 denominationTokenSalary; // per second in denomination Token
         uint256 accruedValue;
         uint64 lastPayroll;
         uint64 endDate;
     }
 
+    IFeed public feed;
     Finance public finance;
     address public denominationToken;
-    IFeed public feed;
     uint64 public rateExpiryTime;
 
-    mapping(uint256 => Employee) private employees;     // employee ID -> employee
-    // Employees start at index 1, to allow us to use employees[0] to check for non-existent address
-    // mappings with employeeIds
     uint256 public nextEmployee;
+    mapping(uint256 => Employee) private employees;     // employee ID -> employee
     mapping(address => uint256) private employeeIds;    // employee address -> employee ID
     mapping(address => bool) private allowedTokens;
```

Comment:
I sometimes think it's better to provide a useful internal interface rather than allowing subclasses to play directly with the storage structure, just in favor of encapsulation... but with smart contracts, as we cannot predict what we may need in a future, I prefer being more flexible, let's make them `internal` then :)